rules/scripts/sival_signature_test.bash

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+set -o pipefail
+
+FILES_TO_CHECK=(@FILES@)
+
+IS_NEGATIVE_TEST=@IS_NEGATIVE_TEST@
+OPENTITANTOOL=@OPENTITANTOOL@
+VERIFY_ARGS=(@VERIFY_ARGS@)
+ECDSA_KEY=@ECDSA_KEY@
+SPX_KEY=@SPX_KEY@
+KEY_ARGS=()
+RESULT_STATUS=0
+
+echo "___Starting signature verification & manifest key verification___"
+
+for file in "${FILES_TO_CHECK[@]}"
+do
+
+  echo "Checking signature of ${file}..."
+
+  # TEST 1: ECDSA key and/or SPX key integrity and signature verification
+
+  # If ECDSA key is provided, include it in the manifest update
+  if [[ ! -z "$ECDSA_KEY" ]]; then
+    KEY_ARGS+=("--ecdsa-key=${ECDSA_KEY}")
+  fi
+  
+  # If SPX key is provided, include it in the manifest update
+  if [[ ! -z "$SPX_KEY" ]]; then
+    KEY_ARGS+=("--spx-key=${SPX_KEY}")
+  fi
+
+  # If there are keys to verify, perform manifest update
+  if [[ ${#KEY_ARGS[@]} -ne 0 ]]; then
+    echo -n "Checking key material in ${file}..."
+        ${OPENTITANTOOL}  --rcfile= image manifest update \
+          "${KEY_ARGS[@]}" \
+          --output=/tmp/manifest_test_key_verify.$$ \
+          ${file}
+
+    # Perform manifest comparison
+    echo "Comparing dummy manifest with original file..."
+
+    if [[ "$IS_NEGATIVE_TEST" == "True" ]]; then
+      cmp ${file} /tmp/manifest_test_key_verify.$$
+
+      # If negative test, check if successful
+      if [[ $? -eq 0 ]]; then
+        echo "Negative test failed."
+        RESULT_STATUS=1
+      else
+        echo "Negative test succeeded."
+      fi
+    # Else not negative test, check if the verification was successful
+    else
+      cmp ${file} /tmp/manifest_test_key_verify.$$
+
+      if [[ $? -ne 0 ]]; then
+        echo "Key material check failed."
+        RESULT_STATUS=1
+      else
+        echo "Key material check succeeded."
+      fi
+    fi
+
+  else
```

Comment:
I think this `else` changes the behavior of this test in a way we don't want (based on the indentation level, this `else` seems to correspond to the `if` that checks the KEY_ARGS above).

- If the user has provided public key material, we want to check that the signed image contains the provided key material.
- The test should _always_ check the signature.